### PR TITLE
Switch Raman Base Reader to Multiformat Reader

### PR DESCRIPTION
**Pull request by RonHildebrandt**
I used the XPS Multiformat reader as base @lukaspie and will need some serious help to finalize this.

I tried to adapt the XPS Scienta TXT reader to enable the right now present eln_data.yaml from a Raman Witec output. As im right now  lost at which point which modifications have to be done, ill focus right now on a parser for the .txt format.

Could be, that my apppraoch is completely incorrect and it is better to set it up new.

### CODE REVIEW

**Comment by RonHildebrandt** [comment]
The current status is, that I can run:

`dataconverter convert --reader raman --nxdl NXraman examples/witec/txt/eln_data_witec_txt.yaml --output witec_txt_output.nxs`
and .nxs file is created, but the file is completly empty.

**Comment by vedina** [comment]
just a note that https://github.com/h2020charisma/ramanchada2 has support for many spectroscopy formats

**Comment by RonHildebrandt** [comment]
> just a note that https://github.com/h2020charisma/ramanchada2 has support for many spectroscopy formats

Ill take a look. Hopefully we could use this to implement more methods faster. Thanks!

File: src/pynxtools_raman/witec/txt/witec_data_model.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Copyright The NOMAD Authors.
```
Comment (lukaspie) [inline src/pynxtools_raman/witec/txt/witec_data_model.py:1]: You _can_ do it like this, but there is no obligation to do so. This just helps me with type checking in XPS, but you don't have to implement it like this.

File: src/pynxtools_raman/witec/txt/witec_mappings.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
+# Copyright The NOMAD Authors.
```
Comment (lukaspie) [inline src/pynxtools_raman/witec/txt/witec_mappings.py:1]: same here

File: src/pynxtools_raman/witec/txt/witec_reader.py [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+from pynxtools_raman.witec.txt.witec_mappings import (
+    UNITS,
+    VALUE_MAP,
+    _construct_date_time,
+    _get_key_value_pair,
+)
+from pynxtools_raman.value_mappers import convert_units, get_units_for_key
+
+
+class MapperWitec(RamanMapper):
```
Comment (lukaspie) [inline src/pynxtools_raman/witec/txt/witec_reader.py:50]: I suggest you try to get rid of this intermediate mapping class if you don't need it. I am only using it in xps because I have many different versions of the parser and I need a uniform interface to them. But again, there's probably many different (better) ways to achieve this.

File: tests/data/PS-or-PET-Raman-Spectrum-2.csv [hunk 4]
```diff
@@ -1,1617 +0,0 @@
-//Exported ASCII-File
```
Comment (lukaspie) [inline tests/data/PS-or-PET-Raman-Spectrum-2.csv:1]: Why are these deleted? If this is actual measurement data, it makes sense to keep it here for testing.

File: src/pynxtools_raman/reader.py [hunk 5]
```diff
@@ -1,300 +1,705 @@
...
+        ]
+
+
+def concatenate_values(value1, value2):
+    """
+    Concatenate two values of same type to be stored
+    in raman_data_dict. Dicts are merged and every other object is
+    appended to a list.
 
-def load_as_pandas_array(my_file, header):
```
Comment (lukaspie) [inline src/pynxtools_raman/reader.py:96]: Do you really want to get rid of this whole funtionality?

File: src/pynxtools_raman/reader.py [hunk 6]
```diff
@@ -1,300 +1,705 @@
...
 
-    if "sep" in header:
-        clean_header["sep"] = header["sep"].encode("utf-8").decode("unicode_escape")
+def _get_channel_vars(data_vars: List[str]):
+    """Get all data vars that containt _chan."""
+    return [data_var for data_var in data_vars if CHAN_COUNT in data_var]
 
-    return clean_header
 
+def _get_scan_vars(data_vars: List[str]):
```
Comment (lukaspie) [inline src/pynxtools_raman/reader.py:76]: Is that something that you need? This is very specific to the XPS data

File: src/pynxtools_raman/reader.py [hunk 7]
```diff
@@ -1,300 +1,705 @@
...
+            existing = [
+                (key, self.raman_data[key], data_dict[key])
+                for key in set(self.raman_data).intersection(data_dict)
+            ]
+
+            self.raman_data = {**self.raman_data, **data_dict}
+            for key, value1, value2 in existing:
+                self.raman_data[key] = concatenate_values(value1, value2)
+
+    def _get_analyser_names(self) -> List[str]:
```
Comment (lukaspie) [inline src/pynxtools_raman/reader.py:432]: this is only used in XPS

File: src/pynxtools_raman/reader.py [hunk 8]
```diff
@@ -1,300 +1,705 @@
...
+                        detectors += [detector_nm]
+        except KeyError:
+            pass
+
+        if not detectors:
+            detectors += ["detector"]
+
+        return list(dict.fromkeys(detectors))
+
+    def process_multiple_entities(self) -> None:
```
Comment (lukaspie) [inline src/pynxtools_raman/reader.py:471]: Do you actually need this?
